content/300-guides/100-prisma-guides/100-prisma-client-transactions-guide.mdx

```diff
@@ -0,0 +1,778 @@
+
+### Nested writes
+
+Prisma's solution to dependent writes is the **nested writes**  feature, which is supported by `create` and `update`. The following nested write creates one user and two blog posts:
+
+```ts
+  const nestedWrite = await prisma.user.create({
+    data: {
+      email: "imani@prisma.io",
+      posts: {
+        create: [
+          { title: "My first day at Prisma" },
+          { title: "How to configure a unique constraint in PostgreSQL" }
+        ]
+      }
+    }
+  })
+```
+
+If any operation fails, Prisma rolls back the entire transaction. Nested writes are not currently supported by top-level bulk operations like `client.user.deleteMany` and `client.user.updateMany`.
+
+#### When to use nested writes
+
+Consider using nested writes if:
+
+* ✔ You want to create two or more records related by ID at the same time (for example, create a blog post and a user)
+* ✔ You want to update and create records related by ID at the same time (for example, change a user's name and create a new blog post)
+
+> **Tip:** If you [pre-compute your IDs, you can choose between a nested write or using the `$transaction` API](#scenario-pre-computed-ids-and-the-transaction-api).
+
+#### Scenario: Sign-up flow
+
+Consider the Slack sign-up flow, which:
+
+1. Creates a team
+2. Adds one user to that team, which automatically becomes that team's administrator
+
+This scenario can be represented by the following schema - note that users can belong to many teams, and teams can have many users (a many-to-many relationship):
+
+```prisma
+model Team {
+  id   Int    @id @default(autoincrement())
+  name String
+  members User[] // Many team members
+}
+
+model User {
+  id    Int    @id @default(autoincrement())
+  email String @unique
+  teams Team[] // Many teams
+}
+```
+
+The most straightforward approach is to create a team, then create and attach a user to that team:
+
+```ts
+// Create a team
+await prisma.team.create({
+  data: {
+    name: 'Aurora Adventures',
+  },
+})
+
+// Create a user and assign them to the team
+await client.user.create({
+  data: {
+    email: 'alice@prisma.io',
+    team: {
+      connect: {
+        id: team.id,
+      },
+    },
+  },
+})
+```
+
+However, this code has a problem - consider the following scenario:
+
+1. Creating the team succeeds - "Aurora Adventures" is now taken
+2. Creating and connecting the user fails - the team "Aurora Adventures" exists, but has no users
+3. Going through the sign-up flow again and attempting recreate "Aurora Adventures" fails - the team already exists
+
+Creating a team and adding a user should be one atomic operation that **succeeds or fails as a whole**.
+
+To implement atomic writes in a low-level database clients, you must wrap your inserts in `BEGIN`, `COMMIT` and `ROLLBACK` queries. Prisma Client solves the problem with nested writes. The following query creates a team, creates a user, and connects the records in a single transaction:
```

Comment:
maybe link nested writes to the docs corresponding docs page here: https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#nested-writes